pilot/pkg/bootstrap/server.go

```diff
@@ -1191,3 +1204,56 @@ func (s *Server) waitForCacheSync(stop <-chan struct{}) bool {
 
 	return true
 }
+
+func (s *Server) initCertController(args *PilotArgs) error {
+	var err error
+	var svcNames []string
+	var svcNamespaces []string
+	if !s.mesh.K8SCertificateSetting.Enabled {
+		log.Info("k8s certificate provision is not enabled")
+		return nil
+	}
+
+	k8sClient := s.kubeClient
+	if len(s.mesh.K8SCertificateSetting.PilotService) > 0 {
+		// Generate a key and certificate for Pilot and save it
+		// a directory.
+		pilotSvcNs := strings.Split(s.mesh.K8SCertificateSetting.PilotService, ".")
```

Comment:
Users can set `istio-pilot.istio-system.svc.cluster.local`

Reply:
The format of s.mesh.K8SCertificateSetting.PilotService is described in https://github.com/istio/api/pull/1105/: 
  // The Pilot service that will be provisioned with a certificate.
  // E.g., istio-pilot.istio-system.
  // The format is service-name.service-namespace, with '.' as the delimiter.
  // If it is an empty, no Pilot certificate is generated.
  string pilot_service = 3;